ui-ngx/src/app/modules/home/components/widget/lib/chart/time-series-chart-widget.component.ts

```diff
@@ -170,5 +170,9 @@ export class TimeSeriesChartWidgetComponent implements OnInit, OnDestroy, AfterV
 
   public toggleLegendKey(legendKey: LegendKey) {
     this.timeSeriesChart.toggleKey(legendKey.dataKey);
+    for (const id of Object.keys(this.ctx.subscriptions)) {
+      const subscription = this.ctx.subscriptions[id];
+      subscription.updateDataVisibility(legendKey.dataIndex);
+    }
```

Comment:
This method is used for TbFlot. For a better fix, please be sure to look for a problem in timeSeriesChart.toggleKey.